workflows/main.tf

```diff
@@ -18,7 +18,7 @@ resource "aws_lambda_layer_version" "lambda_dependencies" {
   layer_name       = "${local.prefix}-lambda_dependencies"
   source_code_hash = filebase64sha256("${var.DIST_DIR}/lambda_dependencies_layer.zip")
 
-  compatible_runtimes = ["python3.8"]
+  compatible_runtimes = ["python3.8", "python3.9", "python3.10"]
```

Comment:
I think I would limit this to only `python3.10`. There is no guarantee that a package installed on one version will work for another version.

There are several reasons. 
- First of all, the addition of new features on later python versions. So if you install a package on 3.10 for example that uses builtins for type hints such as `list[str]`, that package will not work on 3.8 as that feature wasn't available yet.
- Secondly, if you have any dependencies that include compiled modules, they will only work for that version of python. See the notes on ABI compatibility in the official docs: https://docs.python.org/3/c-api/stable.html#stable. 
  > So, code compiled for Python 3.10.0 will work on 3.10.8 and vice versa, but will need to be compiled separately for 3.9.x and 3.11.x.

  There is a way to compile binaries so that they work across minor versions, but it does require the module to have been written in such a way that it supports that, and I believe you need to use the `--abi` flag when installing the package to tell pip to install the cross compatible version rather than defaulting to the minor version specific one. We aren't doing that in our makefile.